immune_subtype_clustering/prepare_data_sheet.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+from typing import List
+
+import pandas as pd
+import synapseclient
+from synapseclient import File
+
+
+def syn_login():
+    syn = synapseclient.Synapse()
+    syn.login()
+    return syn
+
+
+def download_data_files(parent: str, syn: synapseclient.Synapse):
+    """downloads all TSV files in parent synapse folder from synapse id provided"""
+    children = syn.getChildren(parent=parent)
+    file_entity_list = []
+    for child in children:
+        if child["name"].endswith(".tsv"):
+            file = syn.get(child["id"])
+            file_entity_list.append(file)
+    return file_entity_list
+
+
+def transform_df(df: pd.DataFrame, patient_id: str) -> pd.DataFrame:
```

Comment:
This brings me to the discussion point, this is a pretty minor transformation as glue code for our workflows, if we stay true to the orchestration model, do we have to farm this off to a separate service that does the transforms when using Airflow.  Or.. do you think small things like this is fine to be done within Airflow.

@BrunoGrandePhD

Reply:
I started the discussion in dpe-topics, so we don't keep this PR blocked by this discussion.